src/gpu-compute/GPU.py

```diff
@@ -96,6 +96,14 @@ class VectorRegisterFile(RegisterFile):
     cxx_header = "gpu-compute/vector_register_file.hh"
 
 
+class RegisterFileCache(SimObject):
+    type = "RegisterFileCache"
+    cxx_class = "gem5::RegisterFileCache"
+    cxx_header = "gpu-compute/register_file_cache.hh"
+    simd_id = Param.Int(-1, "SIMD ID associated with this Register File Cache")
```

Comment:
Just curious why -1 for the default here?

Reply:
I just copied from the RegisterFile above which sets it to -1.

Reply:
IMO, either you should have *no* default, which means the user must set the value, or you should have a sensible default.

If you need to have an id that's unique then you should do this in `startup` (e.g., with a `register` function that calls to the parent object) not expect the user to set the correct ids.

Reply:
I suppose this is a point of debate then -- if @KaiBatley is simply using the same approach that already exists for the GPU, asking him to change it is tricky because then the code itself wouldn't be consistent across the pieces.  But at the same time I can understand not wanting to perpetuate a design one thinks is problematic.

@KaiBatley do you feel like you know what needs to be done to address @powerjg 's request?

Reply:
> simply using the same approach that already exists for the GPU, asking him to change it is tricky because then the code itself wouldn't be consistent across the pieces. But at the same time I can understand not wanting to perpetuate a design one thinks is problematic.

I agree! 

There are two different points/issues here:
1. Default argument `-1` vs. no default. If it is *required* that the user specify something other than `-1`, then I would advocate for making it have no default so there is an error if the user doesn't set the value. If, on the other hand, `-1` is meaningful, then keeping that the default is OK.
2. Counting "ids" should be done in C++ (or python) not *by the user*. This is a broader design problem that is pervasive across not just the GPU but also Ruby. This is something that should probably be addressed in a separate commit, and it not particularly important.

My comment was more about (1) than (2). (2) was more for @KaiBatley (and others') benefit in the future.

Reply:
I have removed the default of -1 for the simd_id as it has no meaningful value.

Reply:
Thanks!